### PR TITLE
feat: migrate to PDM

### PR DESCRIPTION
**Pull request by JacobCoffee**
[//]: # "By submitting this pull request, you agree to:"
[//]: # "- follow [Litestar's Code of Conduct](https://github.com/litestar-org/.github/blob/main/CODE_OF_CONDUCT.md)"
[//]: # "- follow [Litestar's contribution guidelines](https://github.com/litestar-org/.github/blob/main/CONTRIBUTING.md)"

### Pull Request Checklist

- [x] New code has 100% test coverage
- [x] (If applicable) The prose documentation has been updated to reflect the changes introduced by this PR
- [x] (If applicable) The reference documentation has been updated to reflect the changes introduced by this PR
- [x] Pre-Commit Checks were ran and passed
- [x] Tests were ran and passed

### Description
[//]: # "Please describe your pull request for new release changelog purposes"

- move to pdm
- move things into `pyproject.toml` 
- small changes
- add new repo rename script to swap all `project-template` instances to `${repo_name}`

### Close Issue(s)
[//]: # "Please add in issue numbers this pull request will close, if applicable"
[//]: # "Examples: Fixes #4321 or Closes #1234"

-

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,26 +1,115 @@
...
+.PHONY: lint
+lint: 												## Runs pre-commit hooks; includes ruff linting, codespell, black
+	@echo "=> Running pre-commit process"
+	@$(ENV_PREFIX)pre-commit run --all-files
+	@echo "=> Pre-commit complete"
+
+.PHONY: coverage
+coverage:  											## Run the tests and generate coverage report
+	@echo "=> Running tests with coverage"
+	@$(ENV_PREFIX)pytest tests --cov=app
```
Comment (cofin) [inline Makefile]: I think we need to set the test path to `src/`?

**Review by cofin — DISMISSED** [review]

File: .github/workflows/publish.yaml [hunk 2]
```diff
@@ -1,23 +1,28 @@
...
+
+      - uses: pdm-project/setup-pdm@v3
+        name: Set up PDM
+        with:
+          python-version: "3.11"
+          allow-python-prereleases: true
+          cache: true
+
+      - name: Publish package distributions to PyPI
+        run: pdm publish
```
Comment (provinzkraut) [inline .github/workflows/publish.yaml]: I think we want to use the trusted publishers workflow here

**Review by provinzkraut — DISMISSED** [review]

**Review by cofin — APPROVED** [review]

File: Makefile [hunk 3]
```diff
@@ -1,26 +1,115 @@
...
+	@echo "=> Dependencies Updated"
+	@$(ENV_PREFIX)pre-commit autoupdate
+	@echo "=> Updated Pre-commit"
+
+# =============================================================================
+# Developer Utils
+# =============================================================================
+.PHONY: install-pdm
+install-pdm: 										## Install latest version of PDM
+	@curl -sSL https://pdm.fming.dev/dev/install-pdm.py | python3 -
```
Comment (vkcku) [inline Makefile:34]: This URL is incorrect. It should be:

`curl -sSL https://pdm.fming.dev/install-pdm.py`

File: Makefile [hunk 4]
```diff
@@ -1,26 +1,115 @@
...
+	@if [ "$(USING_PDM)" ]; then $(PDM) update; fi
+	@echo "=> Dependencies Updated"
+	@$(ENV_PREFIX)pre-commit autoupdate
+	@echo "=> Updated Pre-commit"
+
+# =============================================================================
+# Developer Utils
+# =============================================================================
+.PHONY: install-pdm
+install-pdm: 										## Install latest version of PDM
```
Comment (vkcku) [inline Makefile:33]: I think it's a good idea to add a checksum verification as recommended in the installation guide by PDM.

File: Makefile (hunk 3 above)
Comment (JacobCoffee) [reply Makefile:34]: Fixed, thanks - just copied from another project assuming it worked 😅

**Pull request merged**